cranelift/codegen/src/isa/aarch64/inst.isle

```diff
@@ -2961,13 +2967,18 @@
 
 (decl aarch64_link () Reg)
 (rule (aarch64_link)
+      (if (preserve_frame_pointers))
       (if (sign_return_address_disabled))
-      (mov_preg (preg_link)))
+      (let ((dst WritableReg (temp_writable_reg $I64))
+            (addr AMode (AMode.FPOffset 8 $I64))
+            (_ Unit (emit (MInst.ULoad64 dst addr (mem_flags_trusted)))))
+           dst))
 
 (rule (aarch64_link)
-      ;; This constructor is always used for non-leaf functions, so it is safe
-      ;; to clobber LR.
-      (let ((_ Unit (emit (MInst.Xpaclri))))
+      (if (preserve_frame_pointers))
+      (let ((addr AMode (AMode.FPOffset 8 $I64))
+            (_ Unit (emit (MInst.ULoad64 (writable_link_reg) addr (mem_flags_trusted))))
```

Comment:
Can we add a comment here describing the reload? I think this is so that we know that `LR` is in a consistent state (the original signed return address), even if another `get_return_address` op had already been executed; is that right? And then it's safe to clobber below with `xpaclri` because `preserve_frame_pointers` implies that `LR` will be reloaded before return?

Reply:
Actually the main issue is not other `get_return_address` operations, but rather function calls - I added the documentation.